client/packages/react/src/next-ssr/getUserOnServer.ts

```diff
@@ -0,0 +1,8 @@
+import { cookies } from 'next/headers.js';
+
+export const getUserOnServer = async (appId: string) => {
```

Comment:
How about we call this `getUserFromInstantCookie`? 

Reasoning: 

`getUserOnServer` feels a bit awkward because: (a) it doesn't specify what kind of user this is. (b) the server part is understandably important, but it's unclear why it has to be on the server.